File: DS/Binary_Search.py
def binary_search(array,number):
    count=0
    while len(array)>=1:
        mid=int((len(array))/2)
        if(number)==array[mid]:
            mid=number
            count+=1
            break
        elif (number)>array[mid]:
            array=array[mid:]           
        elif(number)<array[mid]:
            array=array[:mid]
        count+=1  
        print(array)
    return(mid,count)

binary_search([0,1,2,3,4,5,6,7,8,9,10],10)

File: Data Structures/C++/binary_search.cpp
#include <iostream>

using namespace std;


void binary_search(int *, int  , int ) ;

int main()
{
    int len;
    
    int a[]={1,3,5,7,9,11}; 
     
    len=sizeof(a)/sizeof(a[0]);
    
    binary_search(a,len,5);
 
   
   
   return 0;
}


void binary_search(int *arr , int length , int target) 
{
    int l=0;
    int h=length-1;
    int count=0;
    
    while(l<=h)
    {
      int middle=(l+h)/2;
      
      if(arr[middle]== target)
      {
          cout<<"位置:"<<middle<<endl;
          cout<<"次數:"<<count+1<<endl;
          return;
      }
      else if(arr[middle] >target)
      {
          h=middle-1;
          count+=1;
      }
      else
      {
          l=middle+1;
          count+=1;
      }
    }    
    
}
